File: Technology Transfer _ Sulambi VMS/Source Code/sulambi-frontend-main/sulambi-frontend-main/src/api/accounts.ts
import axios from "./init";

const basePath = "/accounts";

export const getAllAccounts = () => {
  return axios.get(`${basePath}/`);
};

export const getAllAdminAccounts = () => {
  return axios.get(`${basePath}/admin`);
};

export const getAllOfficerAccounts = () => {
  return axios.get(`${basePath}/officer`);
};

export const createNewAccount = (
  accountType: "admin" | "officer",
  data: { username: string; password: string }
) => {
  return axios.post(`${basePath}/${accountType}`, data);
};

export const deleteAccount = (id: number) => {
  return axios.delete(`${basePath}/${id}`);
};

export const updateAccount = (
  id: number,
  data: { username: string; password: string }
) => {
  return axios.put(`${basePath}/${id}`, data);
};

File: SulambiVosaMain/Technology Transfer _ Sulambi VMS/Source Code/sulambi-backend-main/sulambi-backend-main/app/controllers/accounts.py
from ..models.AccountModel import AccountModel
from flask import request

AccountDb = AccountModel()

def createAccount(accountType):
  username = (request.json.get("username") or "").strip()
  password = (request.json.get("password") or "").strip()
  if not username or not password:
    return ({ "message": "Username and password are required" }, 400)

  matched = AccountDb.getOrSearch(["username"], [
    username
  ])

  if (len(matched) > 0):
    return ({ "message": "Account already exists" }, 403)

  createdAccount = AccountDb.create(
    username,
    password,
    accountType
  )

  return {
    "message": "Account Successfully created",
    "data": createdAccount
  }

def getAccounts(accountType):
  if (accountType == "admin" or accountType == "officer"):
    return {
      "data": AccountDb.getOrSearch(
        ["accountType", "id", "username", "password", "membershipId"],
        [accountType, None, None, None, None]),
      "message": "Successfully retrieved accounts"
    }

  return {
    "data": AccountDb.getAll(),
    "message": "Successfully retrieved accounts"
  }

def deleteAccount(accountId):
  matchedAccount = AccountDb.get(accountId)
  if (matchedAccount == None):
    return ({ "message": "Account id specified does not exist" }, 404)

  AccountDb.delete(accountId)
  return {
    "message": "Successfully deleted account",
    "data": matchedAccount
  }

def updateAccount(accountId):
  matchedAccount = AccountDb.get(accountId)
  if (matchedAccount == None):
    return ({ "message": "Account id specified does not exist" }, 404)

  username = (request.json.get("username") or "").strip()
  password = (request.json.get("password") or "").strip()
  if not username or not password:
    return ({ "message": "Username and password are required" }, 400)

  AccountDb.updateSpecific(accountId, ["username", "password"], (
    username,
    password
  ))

  return {
    "message": "Successfully updated account",
    "data": AccountDb.get(accountId)
  }